db-scheduler/src/test/java/com/github/kagkarlsson/scheduler/scrolling/ScrollingTestBase.java

```diff
@@ -0,0 +1,336 @@
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import javax.sql.DataSource;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public abstract class ScrollingTestBase {
+
+  @RegisterExtension public StopSchedulerExtension stopScheduler = new StopSchedulerExtension();
+
+  protected SchedulerClient schedulerClient;
+  protected SchedulerClient prioritySchedulerClient;
+  private OneTimeTask<Void> task;
+
+  public abstract DataSource getDataSource();
+
+  protected void performDatabaseSpecificSetup() {
+    DbUtils.clearTables(getDataSource());
+  }
+
+  @BeforeEach
+  void setUp() {
+    performDatabaseSpecificSetup();
+
+    task = TestTasks.oneTime("testTask", Void.class, TestTasks.DO_NOTHING);
+
+    Scheduler scheduler =
+        Scheduler.create(getDataSource(), task)
+            .threads(1)
+            .pollingInterval(java.time.Duration.ofSeconds(1))
+            .build();
+    schedulerClient = SchedulerClient.Builder.create(getDataSource(), task).build();
+
+    Scheduler priorityScheduler =
+        Scheduler.create(getDataSource(), task)
+            .enablePriority()
+            .threads(1)
+            .pollingInterval(java.time.Duration.ofSeconds(1))
+            .build();
+    prioritySchedulerClient =
+        SchedulerClient.Builder.create(getDataSource(), task).enablePriority().build();
+
+    stopScheduler.register(scheduler, priorityScheduler);
+  }
+
+  @Test
+  void testBasicScrolling() {
+    createTestExecutions(5);
+
+    List<ScheduledExecution<Void>> firstPage = new ArrayList<>();
+    schedulerClient.fetchScheduledExecutionsForTask(
+        "testTask",
+        Void.class,
+        ScheduledExecutionsFilter.all().withPicked(false).limit(2),
+        firstPage::add);
+
+    assertEquals(2, firstPage.size());
+
+    ExecutionBoundary boundary = ExecutionBoundary.from(firstPage.get(1));
+    List<ScheduledExecution<Void>> secondPage = new ArrayList<>();
+    schedulerClient.fetchScheduledExecutionsForTask(
+        "testTask",
+        Void.class,
+        ScheduledExecutionsFilter.all().withPicked(false).after(boundary).limit(2),
+        secondPage::add);
+
+    assertEquals(2, secondPage.size());
+    assertNotEquals(
+        firstPage.get(0).getTaskInstance().getId(), secondPage.get(0).getTaskInstance().getId());
+  }
+
+  @Test
+  void testBackwardScrolling() {
+    createTestExecutions(5);
+
+    List<ScheduledExecution<Void>> allExecutions = new ArrayList<>();
+    schedulerClient.fetchScheduledExecutionsForTask(
+        "testTask",
+        Void.class,
+        ScheduledExecutionsFilter.all().withPicked(false),
+        allExecutions::add);
+
+    ExecutionBoundary boundary = ExecutionBoundary.from(allExecutions.get(3));
+    List<ScheduledExecution<Void>> beforeResults = new ArrayList<>();
+    schedulerClient.fetchScheduledExecutionsForTask(
```

Comment:
You can test against `getScheduledExecutionsForTask` instead, to avoid the clunkiness of the fetch-variants

Reply:
Can probably save some boilerplate by extracting a method for

```java
schedulerClient.fetchScheduledExecutionsForTask(
        "testTask",
        Void.class,
        ScheduledExecutionsFilter.all().withPicked(false),
        allExecutions::add);
```

something like
```
var scheduled = getScheduledForTask(filterAll -> filterAll.withPicked(false))
```

Reply:
Yes, extracted into a private function to hide boilerplate